proxy/src/bind.rs

```diff
@@ -209,6 +209,9 @@ where
             &self.ctx,
             &addr,
             ep.metadata().clone(),
+            // TODO: when we can use TLS for client connections, indicate
+            //       whether or not the connection was TLS here.
+            ctx::transport::TlsStatus::Disabled,
```

Comment:
I tried to rebase my connect-side work on top of this. In order to compute the `TlsStatus` here, we 'd have to extract the `tls::ClientConfig` out of the `ClientConfigWatch` at this point, right? That would mean that whenever the configuration changes from `None` to `Some(config)`, we'd have to rebind the stack, I think.